### PR TITLE
fix: provide system prompt to MMLU evaluators

### PR DESCRIPTION
**Pull request by RobotSail**
When we run MMLU evaluations today we do not pass the prompt templates to MMLU
which causes the models to be evaluated without the proper chat templates.
This can lead to models acheiving results that are suboptimal from what they
can truly be. This commit addresses that issue by providing a system prompt
argument to the MMLU evaluator objects.

Signed-off-by: Oleg S <97077423+RobotSail@users.noreply.github.com>

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Comment by RobotSail** [comment]
Tests are currently failing because this PR depends on https://github.com/instructlab/eval/pull/197

**Review by danmcp — COMMENTED** [review]
Changes LGTM, there is also a reference to mmlu here:

https://github.com/instructlab/instructlab/blob/d739ed57de194e5fc2ff48f59c84ff8ab047b0b3/src/instructlab/model/accelerated_train.py#L590

It's a dead code path but should either update, remove, or at least add a comment to it.

**Comment by RobotSail** [comment]
Thanks for catching that @danmcp, I shall update it!

File: src/instructlab/model/evaluate.py [hunk 1]
```diff
@@ -827,20 +833,29 @@ def evaluate(
...
                     model,
                     tasks_dir,
                     ["mmlu_pr"],
                     few_shots=few_shots,
                     batch_size=batch_size,
+                    system_prompt=model_system_prompt,
                 ),
                 MMLUBranchEvaluator(
                     base_model,
+                    base_model_system_prompt,
```
[the remark this answers is not in the document]
Comment (relyt0925) [reply src/instructlab/model/evaluate.py:853]: compare to above evaluator you put in place and you should see what I mean based on the function signature

[the remark this answers is not in the document]
Comment (relyt0925) [reply src/instructlab/model/evaluate.py:853]: error with code as now on second eval
`Tasks dir not found: I am a Red Hat® Instruct Model, an AI language model developed by Red Hat and IBM Research based on the granite-3.0-8b-base model. My primary role is to serve as a chat assistant.`

[the remark this answers is not in the document]
Comment (relyt0925) [reply src/instructlab/model/evaluate.py:853]: Fixed in my branch with this commit:
https://github.com/relyt0925/instructlab/commit/47a652d49d0f59583497914c9cb5cdf4ee318b51

[the remark this answers is not in the document]
Comment (RobotSail) [reply src/instructlab/model/evaluate.py:853]: Good catch, I'll update this to use only keyword-arguments to avoid future confusion

**Comment by relyt0925** [comment]
With that last change: I was able to get successful results for mmlu eval and mtbench eval

I will open an issue that now that we have moved to this the "auto" batch size feature is no longer applicable and you have to explicitly specify the batch size in order to send batch requests.

**Comment by RobotSail** [comment]
> I will open an issue that now that we have moved to this the "auto" batch size feature is no longer applicable and you have to explicitly specify the batch size in order to send batch requests.

I'm under the impression that this is only true when doing evaluation with vLLM, but without vLLM it seems like it works fine with batching. But in the CLI, it's always running through vLLM. We may need to investigate this further.

**Comment by danmcp** [comment]
> I'm under the impression that this is only true when doing evaluation with vLLM, but without vLLM it seems like it works fine with batching. But in the CLI, it's always running through vLLM. We may need to investigate this further.

This is my understanding as well.

**Review by jaideepr97 — APPROVED** [review]
LGTM thanks @RobotSail

**Comment by nathan-weinberg** [comment]
@Mergifyio backport release-v0.21

**Pull request merged**